server/src/main/java/org/elasticsearch/action/support/master/TransportMasterNodeAction.java

```diff
@@ -56,6 +58,8 @@
 public abstract class TransportMasterNodeAction<Request extends MasterNodeRequest<Request>, Response extends ActionResponse>
     extends HandledTransportAction<Request, Response> {
 
+    protected static final Logger logger = LogManager.getLogger(TransportMasterNodeAction.class);
```

Comment:
I wonder if this should be static or not? Never really digged into why most loggers are done non-statically...

Reply:
Ah 🤦‍♂ we don't need this logger ... the one in the parent is non-static so it'll work fine with just the setting. Thanks for spotting this.

Reply:
9d5dca9

Reply:
I think loggers that are fixed to the declaring class like this should be `private static final`. Unfortunately there's a number of subclasses that are now using this logger (and tests are failing because of it) so it's a bit more work to sort it out. However, it is worth it in the long run - it's much easier to work out which logger logs a particular message if you don't have to worry about inheritance.